src/api/firebase.js

```diff
@@ -92,16 +93,77 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 export async function updateItem(listId, itemId, checked) {
 	const itemRef = doc(db, listId, itemId);
 	const itemDoc = await getDoc(itemRef);
-	const { dateLastPurchased, totalPurchases } = itemDoc.data();
+	const { dateCreated, dateLastPurchased, dateNextPurchased, totalPurchases } =
+		itemDoc.data();
 
-	try {
-		return updateDoc(itemRef, {
-			dateLastPurchased: checked ? new Date() : dateLastPurchased,
-			totalPurchases: checked ? totalPurchases + 1 : totalPurchases,
-			checked: checked,
-		});
-	} catch (error) {
-		console.log('updateDoc error', error);
+	const checkedDateLastPurchased = checked ? new Date() : dateLastPurchased;
+
+	if (checkedDateLastPurchased) {
+		// Convert dateNextPurchased from Timestamp to a JS Date
+		const dateNextPurchasedAsDate = dateNextPurchased.toDate();
+
+		/**
+		 * Calculate the previous estimate for purchase
+		 * based on historical data
+		 */
+		const previousEstimate = getDaysBetweenDates(
+			checkedDateLastPurchased,
+			dateNextPurchasedAsDate,
+		);
+
+		/**
+		 * Calculate the days since the last transaction,
+		 * considering either last purchased or the created date
+		 */
+		const daysSinceLastTransaction = getDaysBetweenDates(
+			new Date(),
+			checkedDateLastPurchased || dateCreated,
+		);
+
+		// Calculate the remaining days until the next purchase
+		let remainingDays = calculateEstimate(
```

Comment:
Based on your comments, would this make more sense if it were named `daysTillNextPurchase` ?

Reply:
@mentalcaries agreed, just modified! ✨